### DISCUSSION TITLE
Adding custom command line args not working

### DISCUSSION THREAD

**Discussion by Thundercleez** [pytest-dev/pytest#11035] [Q&A] [answered]
I have my conftest.py as

```
import pytest

def pytest_addoption(parser):
  parser.addoption("--zdebug", type=bool, default=True, action="store")

@pytest.fixture
def zdebug(request):
  return request.config.getoption('--zdebug')
```

Then I have another file foo.py

```
def test_foo(zdebug)
  print("zdebug: " + str(zdebug))
```

Then at the command line I run

`pytest -s --zdebug=False foo.py`

Even if I set the default in the addoption to False, the print out always prints True. Why is this not working?

Of course, I've tried

`pytest -s --zdebug False foo.py`

as well

**Comment by RonnyPfannschmidt** [comment, marked as answer]
As per [the docs on the type arg](https://docs.python.org/3/library/argparse.html#type) bool ist discouraged as type since bool Just Checks for truthyness

If you want a flag i recommend the Store true Action
